pisek/task_jobs/solution.py

```diff
@@ -55,32 +59,71 @@ def _get_jobs(self) -> list[Job]:
         self._compile_job = compile_
 
         self._judges: dict[TaskPath, RunJudge] = {}
-        for sub_num, sub in self._env.config.subtasks.items():
+
+        for sub_num, inputs in self._all_inputs().items():
             self.subtasks.append(SubtaskJobGroup(self._env, sub_num))
-            for inp in self._subtask_inputs(sub):
-                if inp not in self._judges:
-                    run_sol: RunSolution
-                    run_judge: RunJudge
-                    if self._env.config.task_type == "batch":
-                        run_sol, run_judge = self._create_batch_jobs(sub_num, inp)
-                        jobs += [run_sol, run_judge]
-                        self._outputs.append((run_judge.output, run_judge))
-
-                    elif self._env.config.task_type == "communication":
-                        run_sol = run_judge = self._create_communication_jobs(inp)
-                        jobs.append(run_sol)
-
-                    self._judges[inp] = run_judge
-                    self.subtasks[-1].new_jobs.append(run_judge)
-                    self.subtasks[-1].new_run_jobs.append(run_sol)
-                else:
-                    self.subtasks[-1].previous_jobs.append(self._judges[inp])
+            for inp in inputs:
+                jobs += self._input_info_jobs(inp, sub_num)
+
+        return jobs
+
+    def _skip_input(self, input_info: InputInfo, seed: int, subtask: int) -> bool:
+        input_path = input_info.task_path(self._env, seed)
+        if input_path in self._judges:
+            self.subtasks[-1].previous_jobs.append(self._judges[input_path])
+        return super()._skip_input(input_info, seed, subtask)
+
+    def _generate_input_jobs(
+        self, input_info: InputInfo, seed: int, subtask: int, test_determinism: bool
+    ) -> list[Job]:
+        if not self._is_first:
+            return []
+        return super()._generate_input_jobs(input_info, seed, subtask, test_determinism)
+
+    def _respects_seed_jobs(
+        self, input_info: InputInfo, seeds: list[int], subtask: int
+    ) -> list[Job]:
+        if not self._is_first:
+            return []
+        return super()._respects_seed_jobs(input_info, seeds, subtask)
```

Comment:
I don't understand what those methods are for.

Reply:
So we don't generate the jobs again if we are not on first solution.

Jobs handling still requires some not trival running time.